src/Sylius/Bundle/ShopBundle/Controller/LocaleController.php

```diff
@@ -84,13 +76,12 @@ public function renderSelectorAction()
 
     /**
      * @param Request $request
-     * @param string $code
      *
      * @return Response
      */
     public function switchLocaleAction(Request $request, $code)
     {
-        $this->localeStorage->set($this->channelContext->getChannel(), $code);
+        $this->localeChangeHandler->handle($code);
```

Comment:
Deeply in my heart I feel that it should be a command, but I understand that it could change our architecture and we don't want such a changes right now. Maybe atleast it should be a value object. WDYT @pjedrzejewski ?